### PR TITLE
Components: Avoid displaying tooltip on focus from mousedown

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to resolve an issue where a button with an associated tooltip will briefly flash the tooltip when clicked. The underlying issue here is that a tooltip is expected to be shown (immediately) when a button is focused. Previously, this was not disambiguated from focus which results from clicking to interact with the button, where the tooltip is not expected to be shown.

Before|After
---|---
tooltip-before|tooltip

**Testing instructions:**

Verify there are no regressions in the display of the tooltips being shown upon focusing the button.

Verify that the tooltip is not shown when clicking on a button with an associated tooltip (e.g. inserter, block movers, etc).

_Needs Accessibility Feedback:_ I'd appreciate any advice on whether I've overlooked any use cases with the assumption that focus-via-click should be ignored.

### CODE REVIEW

**Comment by gziolo** [comment]
It looks like this change invalidated several snapshots saved for unit tests:
https://travis-ci.com/WordPress/gutenberg/jobs/220649539

I guess `npm run test-unit -- -u` is something that will solve it.

File: packages/components/src/tooltip/index.js [hunk 1]
```diff
@@ -35,13 +35,29 @@ class Tooltip extends Component {
 			TOOLTIP_DELAY
 		);
 
+		/**
```
Comment (mtias) [inline packages/components/src/tooltip/index.js]: Should this be just a multiline comment with `//`?

Comment (aduth) [reply packages/components/src/tooltip/index.js]: > Should this be just a multiline comment with `//`?

It's meant to be JSDoc describing the instance variable. I guess including `@type` (like in the subsequent one) could help make this clearer).

**Comment by aduth** [comment]
> It looks like this change invalidated several snapshots saved for unit tests:
> https://travis-ci.com/WordPress/gutenberg/jobs/220649539
> 
> I guess `npm run test-unit -- -u` is something that will solve it.

Thanks. Updated in 3363cb627.

**Review by gziolo — APPROVED** [review]
Code looks good and it works as advertised in my testing 👍 

I didn't discover any regressions when using keyboard only and tabbing between icon-only buttons.

**Pull request merged**